zenoh/src/net/routing/dispatcher/resource.rs

```diff
@@ -345,7 +349,7 @@ impl Resource {
 
     #[cfg(test)]
     pub fn print_tree(from: &Arc<Resource>) -> String {
-        let mut result = from.expr();
+        let mut result = from.expr().to_string();
```

Comment:
I agree with the change of having `expr() -> &str`. I'm not sure `expr_to_string() -> String` is really needed since, like here, you can easily do `expr().to_string()`.

Reply:
I agree, `expr_to_string()` is a pure, non very useful, syntaxic shortcut. The main reason fo it was easier query replace and easier discrimination of places we need an owned string from places where we don't.
It can be removed.